model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/JpaUpdate19_0_0_DefaultPostLogoutRedirectUri.java

```diff
@@ -17,32 +17,21 @@
 package org.keycloak.connections.jpa.updater.liquibase.custom;
 
 import liquibase.exception.CustomChangeException;
-import liquibase.statement.core.InsertStatement;
-import liquibase.structure.core.Table;
-
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
+import liquibase.statement.core.RawSqlStatement;
 
 public class JpaUpdate19_0_0_DefaultPostLogoutRedirectUri extends CustomKeycloakTask {
 
     private static final String POST_LOGOUT_REDIRECT_URIS = "post.logout.redirect.uris";
 
     @Override
     protected void generateStatementsImpl() throws CustomChangeException {
-        String sql = "SELECT DISTINCT CLIENT_ID FROM " + getTableName("REDIRECT_URIS");
-
-        try (PreparedStatement statement = jdbcConnection.prepareStatement(sql); ResultSet rs = statement.executeQuery()) {
-            while (rs.next()) {
-                statements.add(
-                        new InsertStatement(null, null, database.correctObjectName("CLIENT_ATTRIBUTES", Table.class))
-                                .addColumnValue("CLIENT_ID", rs.getString(1))
-                                .addColumnValue("NAME", POST_LOGOUT_REDIRECT_URIS)
-                                .addColumnValue("VALUE", "+")
-                );
-            }
-        } catch (Exception e) {
-            throw new CustomChangeException(getTaskId() + ": Exception when extracting data from previous version", e);
-        }
+        final String clientAttributesTable = getTableName("CLIENT_ATTRIBUTES");
+        final String redirectUrisTable = getTableName("REDIRECT_URIS");
+        statements.add(new RawSqlStatement(
+                "INSERT INTO " + clientAttributesTable + " (CLIENT_ID,NAME,VALUE) " +
+                "SELECT DISTINCT CLIENT_ID, '" + POST_LOGOUT_REDIRECT_URIS + "', '+' FROM " + redirectUrisTable + " WHERE CLIENT_ID NOT IN " +
+                "(SELECT CLIENT_ID FROM " + clientAttributesTable + " WHERE NAME = '" + POST_LOGOUT_REDIRECT_URIS + "')"
```

Comment:
The intention of this command is correct.

However the syntax may vary slightly per database vendor. Has this command been tried on all supported databases?

Reply:
@hmlnarik Not in all (I personally tested in H2, mariadb/mysql and postgresql), but I have checked there similar SQLs in other upgrade classes like [here](https://github.com/keycloak/keycloak/blob/21.0.2/model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/JpaUpdate13_0_0_MigrateDefaultRoles.java#L68-L71). I can check in MSSQL too but I cannot test in Orcale (but I'm almost sure this is going to work in both of them).

Reply:
Tested in MSSQL 2022 too, it works. So only oracle is missing. But I think that the syntax is OK with Oracle.

Reply:
@rmartinc I've sent you some notes (by email) how to test with Oracle docker image

In general, it will be better if we can rely on automated tests, and pipeline. However at the same time, I understand that sometimes it is safer to test things manually (also not sure about our pipeline state and if/how is possible to test migration here).

Reply:
Thanks!

I've already approved this PR and from my point of view, it is good to go.

@hmlnarik WDYT?